src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,55 @@
+package mate.academy.service.impl;
+
+import jakarta.persistence.EntityNotFoundException;
+import java.util.Collections;
+import java.util.Optional;
+import mate.academy.dao.ShoppingCartDao;
+import mate.academy.dao.TicketDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.MovieSession;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.Ticket;
+import mate.academy.model.User;
+import mate.academy.service.ShoppingCartService;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+    @Inject
+    private TicketDao ticketDao;
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        Ticket newTicket = new Ticket();
+        newTicket.setUser(user);
+        newTicket.setMovieSession(movieSession);
+        ticketDao.add(newTicket);
+        Optional<ShoppingCart> optionalFromDB = shoppingCartDao.getByUser(user);
```

Comment:
```suggestion
        ShoppingCart shoppingCartFromDB = getByUser(user);
```

you can use method from service ShoppingCart getByUser(User user)

Reply:
not fixed